File: tarea4/cpp/vaciarMatriz.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(int argc, char* argv[]) {
    vector<vector<int>> matriz = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9}
    };
    vector<int> arreglo;

    if (argc == 1) {
        for (auto &fila : matriz) {
            cout << "[";
            for (size_t i = 0; i < fila.size(); i++) {
                cout << fila[i] << (i < fila.size() - 1 ? ", " : "");
            }
            cout << "] ";
        }
        cout << endl;
    } else {
        string comando = argv[1];
        if (comando == "filas") {
            for (auto &fila : matriz) {
                for (int elemento : fila) {
                    arreglo.push_back(elemento);
                }
            }
        } else if (comando == "columnas") {
            for (size_t i = 0; i < matriz[0].size(); i++) {
                for (size_t j = 0; j < matriz.size(); j++) {
                    arreglo.push_back(matriz[j][i]);
                }
            }
        }

        for (size_t i = 0; i < arreglo.size(); i++) {
            cout << arreglo[i] << (i < arreglo.size() - 1 ? " " : "\n");
        }
    }
}

File: tarea4/vaciarMatriz.java
import java.util.*;

public class vaciarMatriz {
    public static void main(String[] args) {
        int[][] matriz = {
            {1, 2, 3},
            {4, 5, 6},
            {7, 8, 9}
        };

        ArrayList<Integer> arreglo = new ArrayList<>();

        if (args.length == 0) {
            System.out.println(Arrays.deepToString(matriz));
        } else {
            switch (args[0]) {
                case "filas": {
                    for (int[] fila : matriz) {
                        for (int elemento : fila) {
                            arreglo.add(elemento);
                        }
                    }
                    System.out.println(arreglo);
                    break;
                }
                case "columnas": {
                    for (int i = 0; i < matriz[0].length; i++) {
                        for (int j = 0; j < matriz.length; j++) {
                            arreglo.add(matriz[j][i]);
                        }
                    }
                    System.out.println(arreglo);
                    break;
                }
                default:
                    System.out.println("Comando no reconocido");
            }
        }
    }
}